contracts/shared/ERC20Permit.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.17;
+
+import "../interfaces/IERC20Permit.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+/**
+ * @dev Implementation of the ERC20 Permit extension allowing approvals to be made via signatures, as defined in
+ * https://eips.ethereum.org/EIPS/eip-2612[EIP-2612].
+ *
+ * Adds the {permit} method, which can be used to change an account's ERC20 allowance (see {IERC20-allowance}) by
+ * presenting a message signed by the account. By not relying on `{IERC20-approve}`, the token holder account doesn't
+ * need to send a transaction, and thus is not required to hold Ether at all.
+ *
+ */
+abstract contract ERC20Permit is ERC20, IERC20Permit {
```

Comment:
pls consider using https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.8.0/contracts/token/ERC20/extensions/draft-ERC20Permit.sol instead of this custom contract. 
it is tested, verified and audited.
the generic contract is initialized with the values needed for permitted allowance when deployed.  
it will only require "DLLR" for the name and "1" for version in the constructor for the DLLR (TBD).

Reply:
pls consider using https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.8.0/contracts/token/ERC20/extensions/draft-ERC20Permit.sol instead of this custom contract. 
it is tested, verified and audited.
the generic contract is initialized with the values needed for permitted allowance when deployed.  
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/7c5f6bc2c8743d83443fa46395d75f2f3f99054a/contracts/utils/cryptography/EIP712.sol#L53 
it will only require "DLLR" for the name and "1" for version in the constructor for the DLLR (TBD).